src/5.0_to_5.1/migrateSettings.ts

```diff
@@ -0,0 +1,73 @@
+import { ContentRecord } from "@activeviam/activeui-sdk-5.0";
+
+const settingsToPermissionsMapping = {
+  "userFilters.areEnabled": "canUseUserFilters",
+  "userQueryContext.isEnabled": "canUseUserQueryContext",
+  "calculatedMeasures.areEnabled": "canUseCalculatedMeasures",
+};
+
+/**
+ * Moves some attributes from `settings` to `permissions` and updates their name.
+ * Only moves an attribute if the corresponding permission doesn't exist yet.
+ * If there is no `permissions`, deletes the `settings` attribute that were supposed to be moved.
```

Comment:
In this case, the migration fails: the settings were not moved anywhere. I'm not sure we can just create a `permissions` folder out of the blue. .
Should we raise an error message to tell the users the migration failed and that they should manually create a permissions folder? And not delete any setting I guess (I did it for now to have a default behavior until we reach a decision).

Reply:
I think just deleting the setting is dangerous. What's the drawback of creating the `permissions` folder for them? If it's a big deal then I would go for the next best & safe thing: raising an error asking the user to create the folder themself and then rerun the migration script.

Reply:
Actually you're right, I think we can just create it manually we some default attributes, I don't think it would cause any drawbacks 🤔
Let me give that a try